src/main/java/com/github/splendor_mobile_game/game/model/Game.java

```diff
@@ -47,6 +48,24 @@ public User changeTurn(){
         }
     }
 
+    public Card takeCardFromRevealed(Card card){
```

Comment:
I would consider splitting this function into two. One that removes from revealed cards and one that draws new card. Or rename this function so it's side effects are clear. Also such functions should have java docs now or another issue will pop up with this task.